src/ccddata.jl

```diff
@@ -14,9 +14,34 @@ Base.promote_rule(::Type{AbstractCCDData{T}}, ::Type{AbstractCCDData{V}}) where
 
 # custom data type to hold ImageHDU
 """
-    CCDData(data::AbstractMatrix, hdr::FITSHeader)
+    CCDData(data::AbstractMatrix, [hdr::FITSHeader])
 
 Struct to store `ImageHDU`, derived from [`AbstractCCDData`](@ref).
+
+`CCDData` acts like a matrix with a header associated.
+
+```julia
+ccd = CCDData(zeros(4, 4))
+
+ccd[1]
+```
+This accesses the 1st element in matrix associated with `ccd`.
+
+One can perform arithmetic operations on it as well:
+
+```julia
+ccd1 = CCDData(zeros(4, 4))
+
+ccd2 = CCDData(ones(4, 4))
+
+sum_ccd1 = ccd1 + ccd2
+
+sum_ccd2 = ccd2 + ccd1
+```
+
+`sum_ccd1` has the header of `ccd1` whereas `sum_ccd2` has the header `ccd2`.
+
+If header is not provided in the `CCDData` constructor, `default_header` is used to generate the header.
```

Comment:
Can we add `(@ref)` to link documentation of other packages?

Reply:
you can't. Typically you can link to the package url or just don't link it.

Reply:
Do we link it like [abc](www.github.com)?

I tried this, but the REPL output for help gets messed up. (the link also becomes visible)

Reply:
by default the docstrings are going to show the full URL, not just a blue-text like you see with `@ref`, so that's the expected output.